### PR TITLE
Merge YP core logic

### PR DESCRIPTION
**Pull request by archiepeterjarvis**

### CODE REVIEW

File: src/asmcnc/job/yetipilot/config/algorithm_parameters.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    },
+    {
+      "Name": "m_coefficient",
+      "Value": 1.0,
+      "Description": "Adjustment algorithm parameter"
+    },
+    {
+      "Name": "c_coefficient",
+      "Value": 35.0,
+      "Description": "Adjustment algorithm parameter"
```
[the remark this answers is not in the document]
Comment (archiepeterjarvis) [reply src/asmcnc/job/yetipilot/config/algorithm_parameters.json:48]: These are specced by Ed & Benji

File: src/asmcnc/job/yetipilot/yetipilot.py [hunk 2]
```diff
@@ -1,59 +1,300 @@
...
         self.sm = kwargs['screen_manager']
         self.jd = kwargs['job_data']
+        self.get_available_profiles()
+        self.load_parameters()
+
+    # System
+    def enable(self):
+        self.use_yp = True
+
+        if self.sm.has_screen('go'):
```
[the remark this answers is not in the document]
Comment (archiepeterjarvis) [reply src/asmcnc/job/yetipilot/yetipilot.py:87]: it works atm

File: src/asmcnc/job/yetipilot/yetipilot.py [hunk 3]
```diff
@@ -1,59 +1,303 @@
...
+    def add_to_stack(self, digital_spindle_ld_qdA, feed_override_percentage, feed_rate, current_line_number,
+                     digital_spindle_mains_voltage):
+        if not self.active_profile and not self.using_advanced_profile:
+            return
+
+        self.digital_spindle_mains_voltage = digital_spindle_mains_voltage
+
+        digital_spindle_ld_w = self.ldA_to_watts(digital_spindle_ld_qdA)
+
+        if len(self.digital_spindle_load_stack) == self.spindle_load_stack_size:
```
[the remark this answers is not in the document]
Comment (archiepeterjarvis) [reply src/asmcnc/job/yetipilot/yetipilot.py:149]: impossible

File: src/asmcnc/job/yetipilot/yetipilot.py [hunk 4]
```diff
@@ -1,59 +1,314 @@
...
+
+            if self.m.s.feed_override_percentage == 200 and adjustment > 0:
+                break
+
+            if self.m.s.feed_override_percentage + adjustment > 200 and adjustment == 10:
+                adjustment = 1
+
+            if self.m.s.feed_override_percentage + adjustment < 10:
+                if not self.waiting_for_feed_too_low_decision:
+                    Clock.schedule_once(lambda dt: self.check_if_feed_too_low(), 4)
```
Comment (LettyYeti) [inline src/asmcnc/job/yetipilot/yetipilot.py:234]: This unchecked clock has potential to throw up a screen at a semi-random time, i.e. if a user has already stopped or paused the job or the spindle has overloaded, then the job & YP could have been disabled but this screen can still be thrown up. Needs some kind of checks that only throw the error if the machine/console is in the right state.

**Review by LettyYeti — CHANGES_REQUESTED** [review]

**Review by LettyYeti — CHANGES_REQUESTED** [review]
Final test before merge

File: src/asmcnc/skavaUI/widget_speed_override.py [hunk 5]
```diff
@@ -160,19 +162,38 @@ def speed_down(self):
...
-            self.sm.get_screen('go').feedOverride.down_5.disabled = False
-            self.sm.get_screen('go').feedOverride.up_5.disabled = False      
-        except:
-            pass
-        self.push = 0
\ No newline at end of file
+
+        if self.m.s.yp.use_yp:
+            if not self.m.s.yp.using_basic_profile:
+                try:
```
Comment (LettyYeti) [inline src/asmcnc/skavaUI/widget_speed_override.py:179]: needs to happen normally as well

**Review by LettyYeti — APPROVED** [review]

**Pull request merged**